packages/aws-cdk/lib/init.ts

```diff
@@ -102,16 +112,18 @@ export class InitTemplate {
    *
    * @param language    the language to instantiate this template with
    * @param targetDirectory the directory where the template is to be instantiated into
+   * @param stackName the output project stack name
+   * @param projectName the output project name
    */
-  public async install(language: string, targetDirectory: string, stackName?: string) {
+  public async install(language: string, targetDirectory: string, stackName?: string, projectName?: string) {
     if (this.languages.indexOf(language) === -1) {
       error(`The ${chalk.blue(language)} language is not supported for ${chalk.green(this.name)} `
           + `(it supports: ${this.languages.map(l => chalk.blue(l)).join(', ')})`);
       throw new Error(`Unsupported language: ${language}`);
     }
 
     const projectInfo: ProjectInfo = {
-      name: decamelize(path.basename(path.resolve(targetDirectory))),
+      name: projectName ?? decamelize(path.basename(path.resolve(targetDirectory))),
```

Comment:
is there any validation we need to do here? for example, are spaces allowed?

Reply:
Nothing is currently stopping a user from doing this:
```sh
mkdir "test folder with spaces"
cd "test folder with spaces"
npx cdk init --language=typescript
```

Which the CDK doesn't handle particularly well out of the box:
`cdk.json`
```json
  "app": "npx ts-node --prefer-ts-exts bin/test folder with spaces.ts",
```

```sh
npm run cdk deploy

> test folder with spaces@0.1.0 cdk
> cdk deploy

node:internal/modules/cjs/loader:1147
  throw err;
  ^

Error: Cannot find module './test'
```

Manually quoting the path in `"app"` does seem to result in the `deploy` and `destroy` working